samples/mpu/mem_domain_apis_test/src/main.c

```diff
@@ -26,13 +26,24 @@ struct k_thread app_thread_id[3];
 struct k_mem_domain app_domain[2];
 
 /* the start address of the MPU region needs to align with its size */
+#ifdef CONFIG_ARM
 u8_t __aligned(32) app0_buf[32];
 u8_t __aligned(32) app1_buf[32];
+#else
+u8_t __aligned(4096) app0_buf[4096];
+u8_t __aligned(4096) app1_buf[4096];
+#endif
+
 
 K_MEM_PARTITION_DEFINE(app0_parts0, app0_buf, sizeof(app0_buf),
 		       K_MEM_PARTITION_P_RW_U_RW);
+#ifdef CONFIG_X86
+K_MEM_PARTITION_DEFINE(app0_parts1, app1_buf, sizeof(app1_buf),
```

Comment:
why do we need different flags for x86?
This should be the same on all arches, please explain your reasoning. we do not want to be going down a path where code that uses memory protection on ARM has to be ifdefed to work with memory protection on x86

Reply:
Andrew in our page tables we cant have the Privileged have write access and the user only read access.
If I set the page table 'U/S' bit,then whatever is in 'R/W' bit will be applicable for both user/supervisor.
I haven't come across a way to provide different access writes to user/supervisor in the same PTE. I think we have discussed this issue before.

Reply:
Ah, now I remember.
Let me think about this some more.

Reply:
this is OK for now